src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -101,6 +102,10 @@ public static bool IsOSPlatform(string platform)
             return platform.Equals("WINDOWS", StringComparison.OrdinalIgnoreCase);
 #elif TARGET_OSX
             return platform.Equals("OSX", StringComparison.OrdinalIgnoreCase) || platform.Equals("MACOS", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_MACCATALYST
+            return platform.Equals("MACCATALYST", StringComparison.OrdinalIgnoreCase) || platform.Equals("IOS", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_IOS
+            return platform.Equals("IOS", StringComparison.OrdinalIgnoreCase);
```

Comment:
I thought we want this:
```suggestion
#elif TARGET_MACCATALYST
            return platform.Equals("MACCATALYST", StringComparison.OrdinalIgnoreCase);
#elif TARGET_IOS
            return platform.Equals("IOS", StringComparison.OrdinalIgnoreCase) || platform.Equals("MACCATALYST", StringComparison.OrdinalIgnoreCase);
```

Reply:
It's correct as it is, allowing "is this IOS" to return `true` on both IOS and MacCatalyst, but "is this MacCatalyst" will only return `true` when on MacCatalyst.

- When running on MacCatalyst
    - `IsOSPlatform("maccatalyst") => true`
    - `IsOSPlatform("ios") => true`
- When running on IOS
    - `IsOSPlatform("ios") => true`
    - `IsOSPlatform("maccatalyst") => false`

Reply:
Never mind I confused 😝